### PR TITLE
Add failover cluster wrapper

### PR DESCRIPTION
**Pull request by madhanrm**

### CODE REVIEW

**Review by raghavendra-nataraj — DISMISSED** [review]

File: pkg/failovercluster/storage/clustersharedvolume/clustersharedvolume_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package clustersharedvolume
```
Comment (raghavendra-nataraj) [inline pkg/failovercluster/storage/clustersharedvolume/clustersharedvolume_test.go:4]: We need a way to test in windows HCI setup

Comment (madhanrm) [reply pkg/failovercluster/storage/clustersharedvolume/clustersharedvolume_test.go:4]: We can use 1es to test these.

**Review by apluchik — DISMISSED** [review]

File: pkg/cluster/compute/affinityrule/affinityrule.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+package affinityrule
+
+import (
+	"github.com/microsoft/wmi/pkg/base/host"
+	"github.com/microsoft/wmi/pkg/base/instance"
+	"github.com/microsoft/wmi/pkg/base/query"
+	"github.com/microsoft/wmi/pkg/constant"
+	wmi "github.com/microsoft/wmi/pkg/wmiinstance"
+	fc "github.com/microsoft/wmi/server2019/root/mscluster"
```
Comment (nikohobart) [inline pkg/cluster/compute/affinityrule/affinityrule.go:12]: Just for my own understanding, I noticed we have pkgs `wmi/server2019` and `wmi/server23h2`. Is there a reason server23h2 is referenced so much less frequently? What's the difference?

**Review by nikohobart — DISMISSED** [review]

**Review by raghavendra-nataraj — APPROVED** [review]

**Pull request merged**